events/bingo/bingoCardApp/ui/js/bingoCard_ui.js

```diff
@@ -0,0 +1,204 @@
+    colorString += ", ";
+    colorString += scriptColorObject.green;
+    colorString += ", ";
+    colorString += scriptColorObject.blue;
+    colorString += ")";
+
+    return colorString;
+}
+
+
+// Handle messages over the EventBridge from bingoCard_app.js
+function onScriptEventReceived(scriptEvent) {
+    if (JSON.parse(scriptEvent).type === "initializeCard") {
+        scriptEvent = JSON.parse(scriptEvent);
+
+        $(".button-number").each(function(index){
+            document.getElementById(index).innerHTML = scriptEvent.allNumbers[index];
+            if (scriptEvent.selectedNumberIDs.indexOf(index) > -1) {
+                document.getElementById(index).classList.add("selected");
+            }
+        });
+
+        // Handle selection of Free space, which has HTML id === -1
+        if (scriptEvent.selectedNumberIDs.indexOf(-1) > -1) {
+            document.getElementById("-1").classList.add("selected");
+        }
+
+        document.getElementById("helpText").innerHTML = " ";
+
+
+        var parsedColor = parseColor(scriptEvent.cardColor);
+        $('body').css("background", parsedColor);
+        
+        $(".button-number").click(function() {
+            if ($(this).hasClass("selected")){
+                $(this).removeClass("selected");
+                EventBridge.emitWebEvent(JSON.stringify({
+                    type: 'bingoNumberDeselected',
+                    deselectedID: parseInt($(this).attr("id"))
+                }));
+            } else {
+                $(this).addClass("selected");
+                EventBridge.emitWebEvent(JSON.stringify({
+                    type: 'bingoNumberSelected',
+                    selectedID: parseInt($(this).attr("id"))
+                }));
+            }
+        });
+
+        $(".button-free").click(function() {
+            if ($(this).hasClass("selected")){
+                $(this).removeClass("selected");
+                EventBridge.emitWebEvent(JSON.stringify({
+                    type: 'bingoNumberDeselected',
+                    deselectedID: parseInt($(this).attr("id"))
+                }));
+            } else {
+                $(this).addClass("selected");
+                EventBridge.emitWebEvent(JSON.stringify({
+                    type: 'bingoNumberSelected',
+                    selectedID: parseInt($(this).attr("id"))
+                }));
+            }
+        });
+
+        var BINGO_STRING = "BINGO";
+        $(".button-letter").click(function() {
+            var index = BINGO_STRING.indexOf($(this).attr('id'));
+            EventBridge.emitWebEvent(JSON.stringify({
+                type: 'playSoundFromBingoHeaderButton',
+                index: index
+            }));
+        });
+
+        $("#bingoButton").click(function() {
+            EventBridge.emitWebEvent(JSON.stringify({
+                type: 'calledBingo'
+            }));
+        });
+    } else if (JSON.parse(scriptEvent).type === "notLoggedIn") {

```

Comment:
It seems like the parse value should be just a general reference at the top to be used in this function

Reply:
all set